tests/ops/qubit/test_matrix_ops.py

```diff
@@ -576,6 +577,114 @@ def test_matrix_already_in_cache_list(self, mat, op):
         assert len(cache["matrices"]) == 3
 
 
+class TestBlockEncode:
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires"),
+        [
+            (1, 1),
+            ([1], 1),
+            ([[1]], 1),
+            (np.array(1), [1]),
+            (np.array([1]), 1),
+            (np.array([[1]]), 1),
+            ([[1, 0], [0, 1]], [0, 1]),
+            (np.array([[1, 0], [0, 1]]), range(2)),
+            (np.identity(3), ["a", "b", "c"]),
+        ],
+    )
+    def test_accepts_various_types(self, input_matrix, wires):
+        op = qml.BlockEncode(input_matrix, wires)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires"),
+        [(1, 1), (1, 2), (1, [1]), (1, range(2)), (np.identity(2), ["a", "b"])],
+    )
+    def test_varied_wires(self, input_matrix, wires):
+        op = qml.BlockEncode(input_matrix, wires)
+
+    @pytest.mark.parametrize(
+        ("input_matrix", "wires", "msg"),
+        [
+            (
+                [[0, 1], [1, 0]],
+                1,
+                f"Block encoding a {2} x {2} matrix requires a hilbert space of size"
```

Comment:
Could we show the matrix shape as (2x2) instead of {2}x{2}?

Reply:
I can add the parentheses, but if I remove the curly brackets from the test code, the regex comparison fails.